### ISSUE TITLE
Issue with delete 

### ISSUE THREAD

**Issue by ncaul** [Cramenorn/RaptorSA#15]
Hey, you had an issue with your delete command, you need to change it to 

`const deleteCount = parseInt(args[1], 10);`

and all of this too.

` if (!message.member.hasPermission("MANAGE_MESSAGES")) {
            message.channel.send("You don't have the permissions to use this command!");
        } else {

            if (isNaN(deleteCount) || deleteCount < 2 || deleteCount > 100) {
                return message.channel.send("Please provide a number between 2 and 100 for the number of messages to delete");
            } else {
               
                message.channel.bulkDelete(deleteCount).catch(error => message.reply(`Couldn't delete messages because of: ${error}`));
               
            }
        }
    }`

Thanks!

**Comment by MalwareWerewolf** [comment]
Hello @Unseencow ,
could you describe what kind of bug you found in the code ?

The line where the constant deleteCount is declared, is correct. I tested your code but it does not work if you do not type a second parameter:

**First test:**

![First Test](https://user-images.githubusercontent.com/43040746/94956946-45d0bc80-04ed-11eb-8e31-31faf776ae0b.PNG)

In the second test I typed **$deletemsg 3 2** and the latest two messages (include the command for the bot) have been deleted.

![Second Test](https://user-images.githubusercontent.com/43040746/94957072-757fc480-04ed-11eb-8284-84daffa4e254.PNG)

The array should start from 0 not 1.

The isNaN function is another possible implementation, but it does not change the code execution:

- First the **$deletemsg** execution:
    ![First the $deletemsg execution](https://user-images.githubusercontent.com/43040746/94957513-29814f80-04ee-11eb-9a3b-04df63bb25fa.PNG)

- After the **$deletemsg** execution:
    ![Capture](https://user-images.githubusercontent.com/43040746/94957785-a9a7b500-04ee-11eb-8202-3a6283546b74.PNG)

**Comment by ncaul** [comment]
Hello Cramenorn


Here is the code that worked for me

 let args = message.content.substring(prefix.length).split(" ");
>     if (message.content.toLowerCase().startsWith(prefix + "clear")) {
>         const deleteCount = parseInt(args[1], 10);
>         if (!message.member.hasPermission("MANAGE_MESSAGES")) {
>             message.channel.send(Embed64);
>         } else {
>
>             if (isNaN(deleteCount) || deleteCount < 1 || deleteCount > 100
> ) {
>                 return message.channel.send(Embed22);
>             } else {
>
>                 message.channel.bulkDelete(deleteCount).catch(error =>
> message.reply(`Couldn't delete messages because of: ${error}`));
>

Without the args[1] it just kept telling me to add a number,

also I added

isNaN(deleteCount)

instead of

>  !deleteCount
>

Why this you ask?

Well, considering the way I have the command written is this:

!clear 10


Meaning 10 messages get deleted, we pass that as the args, (prefix being
arg 0, the number after being arg 1)

If I console.log the args, it returns this:

[image: image.png]

That's how mine worked, maybe your's is structured wrong, or I'm doing it
wrong but this seemed to work for me

Thanks,
Unseen

**Comment by MalwareWerewolf** [comment]
Hello @Unseencow ,
maybe the problem is in another function where the **args** variable is initiliazed.

As I already said your solution is not the best, from the moment you force the user to input an additional parameter, which could be also confusing.

I also tested your solution and sometimes (the first one, I will talk about the second solution below), the command asked me to enter a number even if I entered the number correctly. That's why I said that your solution isn't the best, the problem could be in another function where the bot is initiliazed.

![Capture](https://user-images.githubusercontent.com/43040746/95010726-eaf1af00-062b-11eb-812b-a7dbeaff345b.PNG)

But if this is the case, the issue for this command should appear in other commands, not only in a single one.

There is no need to change the args declaration from const to let:

```
let args = message.content.substring(prefix.length).split(" ");
```

From the moment this variable is initialized one time when you execute a command and it does not change its value. It also appears to be wrong from the moment the command is not doing anything:

![image](https://user-images.githubusercontent.com/43040746/95012209-e2eb3c80-0636-11eb-9386-60027af27f65.png)

**isNaN** as I said is another solution but it does not change the code execution, you can still type a number without getting an error (if it's between 2 and 100 of course). If you type a string it returns an error message:

![image](https://user-images.githubusercontent.com/43040746/95010885-2640ad80-062d-11eb-894c-242b8e4d2c97.png)
 
When you said:

```
Meaning 10 messages get deleted, we pass that as the args, (prefix being
arg 0, the number after being arg 1)
```

The prefix is not inside the args parameter, **args** includes every string separated by a space after the command, in other words a command like **$deletemsg** is not included in the **args**, for example:

```
$deletemsg firstargs secondargs etc.
```

I logged the args variable in message.js and this is what I get (there is no prefix in the array):

![image](https://user-images.githubusercontent.com/43040746/95012773-8b9b9b00-063b-11eb-85af-489e3fddf7ea.png)


I also put a log in the deletemsg command to see the value of the args variable and this is what I get:

```
Args 2
Args 2
Args 6
Args 6
Args 3
Args sdfsdfdfsg
```

I do not get a value like [image: image.png], on the Discord.JS documentation, it specifies that the content property only accepts strings, in this case it can only accepts strings from the args parameter:

![image](https://user-images.githubusercontent.com/43040746/95011460-25aa1600-0631-11eb-95ba-920309663eab.png)

Where do you declare the variables Embed22 and Embed64 in the code ? What is their purpose ?

Your solutions add other functionalities which are not very useful and you are also doing more operations which are not necessary.

In conclusion, no modifications will be applied to the bot.
